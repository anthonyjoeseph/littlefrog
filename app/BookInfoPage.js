// app/BookInfoPage.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image
} from 'react-native';

import Button from 'react-native-button'

import { Actions } from 'react-native-router-flux';

var AImage = require('react-native-image-animation');

class BookInfoPage extends Component {

  constructor(props) {
    super(props)
    this.animationImages = [
      require('./animation/frame_0_delay-0.1s.gif'),
      require('./animation/frame_1_delay-0.1s.gif'),
      require('./animation/frame_2_delay-0.1s.gif'),
      require('./animation/frame_3_delay-0.1s.gif'),
      require('./animation/frame_4_delay-0.1s.gif'),
      require('./animation/frame_5_delay-0.1s.gif'),
      require('./animation/frame_6_delay-0.1s.gif'),
      require('./animation/frame_7_delay-0.1s.gif'),
      require('./animation/frame_8_delay-0.1s.gif'),
      require('./animation/frame_9_delay-0.1s.gif'),
      require('./animation/frame_10_delay-0.1s.gif'),
      require('./animation/frame_11_delay-0.1s.gif'),
      require('./animation/frame_12_delay-0.1s.gif'),
      require('./animation/frame_13_delay-0.1s.gif')
    ];
  }
  render() {
    return (
      <View style={styles.container}>
        <View style={{flex:0.5, justifyContent: 'flex-end'}}>
          <Text style={styles.broccoliLogo}>Little Frog Doesn&#39;t Listen</Text>
          <Text style={styles.title}>Origin: Korea</Text>
        </View>
        <View style={styles.buttons}>
          <View style={{flexDirection:'row', justifyContent:'center', alignItems:'stretch'}}>
            <View style={{flex: 1, flexDirection: 'row', justifyContent: 'flex-end'}}>
            <AImage
                   resizeMode='stretch'
                   animationRepeatCount= {0}
                   animationDuration={100}
                   animationImages={this.animationImages}
                   style={{width:100, height:100}} />
            </View>
            <Text style={{flex: 1, color:'white'}}>This is a traditional story from Korea. Millstone is a rock that grinds the grains into powder. Discover what a magical millstone can do and learn why the ocean is so salty! Remember to flip the page whenever you hear the chime.</Text>
          </View>
          <View style={{flex: 1, flexDirection: 'row', justifyContent: 'flex-end', alignItems: 'flex-end'}}>
            <Button style={{color:'yellow'}} onPress={() => Actions.bookReader()}>Start Reading</Button>
            <Button style={{color:'green'}} onPress={() => Actions.bookReader()}>Read to Me</Button>
          </View>
        </View>
      </View>
    );
  }
}

/*
<AImage
       resizeMode='stretch'
       animationRepeatCount= {0}
       animationDuration={100}
       animationImages={this.animationImages}
       style={{width:null, height:null}} />

*/

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    backgroundColor: '#1586BA',
    flexDirection: 'column',
    height: 100,
    padding: 20
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    color: '#FFFFFF'
  },
  broccoliLogo: {
    fontSize: 40,
    textAlign: 'center',
    color: '#FFFFFF'
  },
  buttons: {
    flex: 0.5,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    flexDirection: 'column'
  }
});

export default BookInfoPage;
