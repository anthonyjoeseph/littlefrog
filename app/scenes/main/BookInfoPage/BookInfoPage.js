// app/BookInfoPage.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image
} from 'react-native';

import Button from 'react-native-button'

import { Actions } from 'react-native-router-flux';

var AImage = require('react-native-image-animation');

class BookInfoPage extends Component {

  constructor(props) {
    super(props)
    this.animationImages = [
      {uri: 'frame1'},
      {uri: 'frame2'},
      {uri: 'frame3'},
      {uri: 'frame4'},
      {uri: 'frame5'},
      {uri: 'frame6'},
      {uri: 'frame7'},
      {uri: 'frame8'},
      {uri: 'frame9'}
    ];
  }
  render() {
    return (
      <View style={styles.container}>
        <View style={{flex:0.5, justifyContent: 'flex-end'}}>
          <Text style={styles.broccoliLogo}>Little Frog Doesn&#39;t Listen</Text>
          <Text style={styles.title}>Origin: Korea</Text>
        </View>
        <View style={styles.buttons}>
          <View style={{flexDirection:'row', justifyContent:'center', alignItems:'stretch'}}>
            <View style={{flex: 1, flexDirection: 'row', justifyContent: 'flex-end'}}>
              <Image
                source={{uri:'spongebob'}}
                style={{width:100, height:100}}
              />
            </View>
            <Text style={{flex: 1, color:'white'}}>This is a traditional story from Korea. Millstone is a rock that grinds the grains into powder. Discover what a magical millstone can do and learn why the ocean is so salty! Remember to flip the page whenever you hear the chime.</Text>
          </View>
          <View style={{flex: 1, flexDirection: 'row', justifyContent: 'flex-end', alignItems: 'flex-end'}}>
            <Button style={{color:'yellow'}} onPress={() => Actions.bookReader()}>Start Reading</Button>
            <Button style={{color:'green'}} onPress={() => Actions.bookReader()}>Read to Me</Button>
          </View>
        </View>
      </View>
    );
  }
}

/*
<AImage
       resizeMode='stretch'
       animationRepeatCount= {0}
       animationDuration={100}
       animationImages={this.animationImages}
       style={{width:100, height:100}} />
*/

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    backgroundColor: '#1586BA',
    flexDirection: 'column',
    height: 100,
    padding: 20
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    color: '#FFFFFF'
  },
  broccoliLogo: {
    fontSize: 40,
    textAlign: 'center',
    color: '#FFFFFF'
  },
  buttons: {
    flex: 0.5,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    flexDirection: 'column'
  }
});

export default BookInfoPage;
