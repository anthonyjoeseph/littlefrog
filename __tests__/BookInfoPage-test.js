// __tests__/BookInfoPage-test.js
import 'react-native';
import React from 'react';
import BookInfoPage from '../app/scenes/main/BookInfoPage/BookInfoPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <BookInfoPage />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
