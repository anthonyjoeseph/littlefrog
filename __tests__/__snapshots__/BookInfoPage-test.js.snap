exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#1586BA",
      "flex": 1,
      "flexDirection": "column",
      "height": 100,
      "justifyContent": "center",
      "padding": 20,
    }
  }>
  <View
    style={
      Object {
        "flex": 0.5,
        "justifyContent": "flex-end",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 40,
          "textAlign": "center",
        }
      }>
      Little Frog Doesn\'t Listen
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "textAlign": "center",
        }
      }>
      Origin: Korea
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.5,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }>
        <Image
          source={
            Object {
              "uri": "spongebob",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          } />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "flex": 1,
          }
        }>
        This is a traditional story from Korea. Millstone is a rock that grinds the grains into powder. Discover what a magical millstone can do and learn why the ocean is so salty! Remember to flip the page whenever you hear the chime.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 17,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              null,
              Object {
                "color": "yellow",
              },
              null,
            ]
          }>
          Start Reading
        </Text>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 17,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              null,
              Object {
                "color": "green",
              },
              null,
            ]
          }>
          Read to Me
        </Text>
      </View>
    </View>
  </View>
</View>
`;
